api/src/test/java/org/pmiops/workbench/actionaudit/auditors/UserServiceAuditorTest.java

```diff
@@ -0,0 +1,79 @@
+package org.pmiops.workbench.actionaudit.auditors;
+
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.verify;
+
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.pmiops.workbench.actionaudit.ActionAuditEvent;
+import org.pmiops.workbench.actionaudit.ActionAuditService;
+import org.pmiops.workbench.actionaudit.AgentType;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.model.DataAccessLevel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Primary;
+import org.springframework.context.annotation.Scope;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+public class UserServiceAuditorTest {
+  static DbUser USER = null;
+
+  @TestConfiguration
+  @Import({UserServiceAuditAdapterImpl.class, ActionAuditTestConfig.class})
+  @MockBean(ActionAuditService.class)
+  static class Configuration {
+    @Primary
+    @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+    public DbUser getUser() {
+      return USER;
+    }
+  }
+
+  @Autowired UserServiceAuditor userServiceAuditor;
+  @Autowired ActionAuditService mockActionAuditService;
+
+  @After
+  public void tearDown() {
+    USER = null;
+  }
+
+  private static DbUser createUser() {
+    final DbUser user = new DbUser();
+    user.setUserId(3L);
+    user.setUsername("foo@gmail.com");
+    return user;
+  }
+
+  @Test
+  public void testFireUpdateDataAccessAction_userAgent() {
+    USER = createUser();
+    userServiceAuditor.fireUpdateDataAccessAction(
+        USER, DataAccessLevel.UNREGISTERED, DataAccessLevel.REGISTERED, AgentType.USER);
+    verify(mockActionAuditService)
```

Comment:
If this isn't working, it might be due to an incompatibility in some of the Mockito stuff. Things like captors return null when the mockito contract requires non-null. There are some examples in the kotlin test files

```
    @Test
    fun testDeleteUserProfile() {
        profileAuditAdapter!!.fireDeleteAction(USER_ID, USER_EMAIL)
        argumentCaptor<ActionAuditEvent>().apply {
            verify(mockActionAuditService).send(capture())
            val eventSent = firstValue

            assertThat(eventSent.targetType).isEqualTo(TargetType.PROFILE)
            assertThat(eventSent.agentType).isEqualTo(AgentType.USER)
            assertThat(eventSent.agentId).isEqualTo(USER_ID)
            assertThat(eventSent.targetIdMaybe).isEqualTo(USER_ID)
            assertThat(eventSent.actionType).isEqualTo(ActionType.DELETE)
//...
```

But this would be the first time we've had this problem in Java AFAIK. Does this work with standard matchers or captors?

 @ericsong thoughts?

Reply:
SOrry for the noise. In reality, my tests were correctly failing - but the error messaging was highly misleading. The arg matchers were working as intended.